crates/trie/sparse/src/trie.rs

```diff
@@ -174,19 +275,25 @@ impl<P: BlindedProvider> SparseTrie<P> {
 /// - All keys in `values` collection are full leaf paths.
 #[derive(Clone, PartialEq, Eq)]
 pub struct RevealedSparseTrie<P = DefaultBlindedProvider> {
-    /// Blinded node provider.
+    /// Provider used for retrieving blinded nodes.
+    /// This allows lazily loading parts of the trie from an external source.
     provider: P,
-    /// All trie nodes.
+    /// Map from a prefix (nibbles) to its corresponding sparse trie node.
```

Comment:
I think "prefix" can be a bit misleading here, I think it's rather a "path"
```suggestion
    /// Map from a path (nibbles) to its corresponding sparse trie node.
```